runtime/syntax/just.vim

```diff
@@ -0,0 +1,400 @@
+" Vim syntax file
+" Language:	Justfile
+" Maintainer:	Noah Bogart <noah.bogart@hey.com>
+" URL:		https://github.com/NoahTheDuke/vim-just.git
+" Last Change:	2024 Aug 05
+
+if exists('b:current_syntax')
+  finish
+endif
+
+let b:current_syntax = 'just'
+
+" syncing fromstart prevents mismatched highlighting when jumping around in a justfile
+" linebreaks= keeps multi-line constructs highlighted correctly while typing
+syn sync fromstart linebreaks=10
+
+" a-zA-Z0-9_-
+syn iskeyword @,48-57,_,-
+
+syn match justComment "#.*$" contains=@Spell,justCommentTodo
+syn match justCommentInBody '#.*$' contained contains=justCommentTodo,justInterpolation,@justOtherCurlyBraces
+syn keyword justCommentTodo TODO FIXME XXX contained
+syn match justShebang "^\s*#!.*$" contains=justInterpolation,@justOtherCurlyBraces
+syn match justName "\h\k*" contained
+syn match justFunction "\h\k*" contained
+
+syn match justPreBodyComment "\v%(\s|\\\n)*%([^\\]\n)@3<!#%([^!].*)?\n%(\t+| +)@=" transparent contained contains=justComment
+   \ nextgroup=@justBodies skipnl
```

Comment:
You are using line-continuation, so you need this part here:
```
  " this file uses line continuations
  let s:cpo_sav = &cpo
  set cpo&vim
```

and then at the end of the file:
```
  let &cpo = s:cpo_sav
  unlet s:cpo_sav
```